### PR TITLE
osslsigncode: add head and test

### PR DESCRIPTION
**Pull request by vszakats**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/osslsigncode.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class Osslsigncode < Formula
   homepage "https://sourceforge.net/projects/osslsigncode/"
   url "https://downloads.sourceforge.net/project/osslsigncode/osslsigncode/osslsigncode-1.7.1.tar.gz"
   sha256 "f9a8cdb38b9c309326764ebc937cba1523a3a751a7ab05df3ecc99d18ae466c9"
+  head "http://git.code.sf.net/p/osslsigncode/osslsigncode"
```
Comment (DomT4) [inline Formula/osslsigncode.rb]: As far as I can see, there's no pre-generated `configure` file in the upstream git repo, so this won't work by itself. You'll get a `Failed to execute ./configure` message or similar.

**Comment by MikeMcQuaid** [comment]
> Have you built your formula locally prior to submission with brew install <formula> (where <formula> is the name of the formula you're submitting)?

Please do this.

**Comment by vszakats** [comment]
Added required head-only logic, added a simple test (this tool works on Windows PE `.exe` or `.msi`  files, so I opted for a `--version` with verified output). Build runs fine now locally.

**Comment by MikeMcQuaid** [comment]
Test fails: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/5696/version=el_capitan/testReport/junit/brew-test-bot/el_capitan/test_osslsigncode/

**Comment by vszakats** [comment]
Updated the test to accommodate for that. Managed to test the test locally, so hopefully it'll be alright this time.

File: Formula/osslsigncode.rb [hunk 2]
```diff
@@ -12,13 +12,26 @@ class Osslsigncode < Formula
...
+    system "autoreconf", "-ivf" if build.head?
     system "./configure", "--prefix=#{prefix}"
     system "make", "install"
   end
+
+  test do
+    # Requires Windows PE executable as input, so we're just showing the version
+    output = `#{bin}/osslsigncode --version`
+    assert_match "osslsigncode", output
+    assert_equal 255, $?.exitstatus
```
Comment (MikeMcQuaid) [inline Formula/osslsigncode.rb]: `assert_match "osslsigncode", shell_output("#{bin}/osslsigncode --version", 255)` does all this a little more idiomatically.

Comment (vszakats) [inline Formula/osslsigncode.rb]: Much nicer indeed, thanks! Updated.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/8244ecb2aa417cc968367917a2bd5f14f52964d7. Thanks again @vszakats! 😺

**Pull request merged**